docs/design/datacontracts/RuntimeTypeSystem.md

```diff
@@ -154,12 +227,39 @@ internal struct MethodTable_1
         EEClassOrCanonMT = data.EEClassOrCanonMT;
         Module = data.Module;
         ParentMethodTable = data.ParentMethodTable;
+        PerInstInfo = data.PerInstInfo;
     }
 }
 ```
 
+Internally the contract depends on a `TypeDescHandle` structure that represents the address of something that implements the TypeDesc data descriptor.
+
+Internally the contract depends on the `TypeHandle` structure being capable of holding a TypeDescHandle or a MethodTableHandle.
+```csharp
+struct TypeHandle
+{
+    internal TypeHandle(TypeDescHandle typeDescHandle);
+    internal TypeHandle(MethodTableHandle methodTableHandle);
+    internal bool IsMethodTable { get; }
+    internal MethodTableHandle AsMethodTable { get; }
+    internal bool IsTypeDesc { get; }
+    internal TypeDescHandle AsTypeDesc { get; }
+}
+```
+
 The contract depends on the global pointer value `FreeObjectMethodTablePointer`.
-The contract additionally depends on the `EEClass` data descriptor.
+The contract additionally depends on these data descriptors
+
+| Data Descriptor Name |
+| --- |
+| `EEClass` |
+| `ArraayClass` |
+| `TypeDesc` |
+| `ParamTypeDesc` |
+| `TypeVarTypeDesc` |
+| `FnPtrTypeDesc` |
+| `GenericsDictInfo` |
+
```

Comment:
In my second pass at this today, I went and did a bit where I put all the types, and all the fields, and comments for the meaning of the fields. It looks really nice, and you guys will see it next week.

Reply:
The other thing we should probably do is put all the magic constants, like enum values in a similar table. Then we can add some build time validation that a various constants have the expected values for the particular contract version we're working with (This could probably even be _DEBUG only validation if its easier to do.). I don't think we need a data descriptor for those values... just enough validation that we know to update the contract version when magic numbers change. I'm not going to do that in this change, but it seems like a good idea for us to implement soonish.